pkg/webui/console/containers/device-payload-formatters/downlink.js

```diff
@@ -65,22 +65,37 @@ class DevicePayloadFormatters extends React.PureComponent {
   static propTypes = {
     appId: PropTypes.string.isRequired,
     devId: PropTypes.string.isRequired,
-    formatters: PropTypes.formatters.isRequired,
+    formatters: PropTypes.formatters,
     updateDevice: PropTypes.func.isRequired,
   }
 
+  static defaultProps = {
+    formatters: undefined,
+  }
+
   async onSubmit(values) {
-    const { appId, devId, formatters, updateDevice } = this.props
+    const { appId, devId, formatters: initialFormatters, updateDevice } = this.props
+
+    if (values.type === PAYLOAD_FORMATTER_TYPES.DEFAULT) {
+      return updateDevice(appId, devId, {
+        formatters: null,
+      })
+    }
```

Comment:
This is not right. That would also erase the other (in this case uplink) payload formatter.

Reply:
This is intentional. You cannot set a particular formatter to null.

Reply:
I see. Hmm, then I'm afraid we need to apply a different UI logic to make this constraint clear for the user. Essentially, instead of using an option in the radio button group, we need a global switch. Something like this:
![image](https://user-images.githubusercontent.com/5710611/80795022-5693a400-8bd6-11ea-8a50-312e6dd24d4b.png)

And maybe, in case of using the default, you can also add a link to the application payload formatters.

Reply:
I agree that such ui is definitely better than the current one, however in order to implement this, quite many changes to our code base are required. Such refactoring can cause regression to functionality that already works well.

Currently, the issue is that the users cannot reset device level payload formatters via the Console. This PR adds this option.

I suggest we discuss possible improvements to the payload formatters pages (both app and device level) in https://github.com/TheThingsNetwork/lorawan-stack/issues/2463 and merge this fix without major refactoring so that in the next release the users have this functionality in place.

Reply:
I see your point, but we're now solving one problem while deliberately introducing another. If we merge it like this, we're knowingly putting users at risk of losing their payload formatter settings. We should at least add some information that setting the formatter to default will erase the formatters altogether.

Reply:
A simple field warning could do the trick here.